src/funcx_common/task_storage/base.py

```diff
@@ -0,0 +1,57 @@
+import abc
+import typing as t
+
+from ..tasks import TaskProtocol
+
+
+class StorageException(Exception):
+    def __init__(self, reason: str):
+        self.reason = reason
+
+    def __str__(self) -> str:
+        return f"Storage request failed due to reason:{self.reason}"
+
+
+class TaskStorage(abc.ABC):
+    """
+    Abstract class which defines the interface for task storage.
+    """
+
+    @abc.abstractmethod
+    def store_result(self, task: TaskProtocol, result: str) -> bool:
+        """
+        Store the result of a task.
+
+        If the storage call succeeded, set reference to data in Task.result_reference
+        and return True
+        If the call failed, an exception would be raised.
+        """
+
+    @property
+    @classmethod
+    @abc.abstractmethod
+    def storage_id(cls) -> str:
+        return NotImplementedError  # type: ignore
```

Comment:
Rather than returning an error class, use `raise NotImplementedError()`.
That will eliminate the need for type-ignore comments, which are a code-smell in a new, completely typed codebase.

Reply:
Fixed.